test/hashmaps/test_maps.fypp

```diff
@@ -142,34 +167,50 @@ contains
     subroutine test_get_data(error, map, test_8_bits, test_block)
         type(error_type), allocatable, intent(out) :: error
         type(chaining_hashmap_type), intent(inout)  :: map
-        integer(int8), intent(in)               :: test_8_bits(test_size)
+        integer(int8), intent(in)               :: test_8_bits(test_size, key_types)
         integer(int_index), intent(in)          :: test_block
         integer :: index2
         type(key_type) :: key
         type(other_type) :: other
         logical :: exists
 
-        do index2=1, size(test_8_bits), test_block
-            call set( key, test_8_bits( index2:index2+test_block-1 ) )
+        do index2=1, test_size, test_block
+            call set( key, test_8_bits( index2:index2+test_block-1, 1 ) )
```

Comment:
Why did you use different columns of `test_8_bits`? Is it not possible to use the same values for all tests?